framework/ChiMesh/LogicalVolume/SurfaceMeshLogicalVolume.cc

```diff
@@ -0,0 +1,213 @@
+#include "SurfaceMeshLogicalVolume.h"
+
+#include "ChiMesh/chi_mesh.h"
+#include "ChiMesh/SurfaceMesh/chi_surfacemesh.h"
+#include "ChiMesh/Raytrace/raytracing.h"
+
+#include "ChiObject/object_maker.h"
+
+#include <utility>
+
+namespace chi_mesh
+{
+
+RegisterChiObject(chi_mesh, SurfaceMeshLogicalVolume);
+
+chi_objects::InputParameters SurfaceMeshLogicalVolume::GetInputParameters()
+{
+  chi_objects::InputParameters params = LogicalVolume::GetInputParameters();
+
+  // clang-format off
+  params.SetGeneralDescription(
+  "\\defgroup chi_mesh__SurfaceMeshLogicalVolume SurfaceMeshLogicalVolume\n"
+  "\\ingroup LuaLogicVolumes");
+  // clang-format on
+
+  params.AddRequiredParameter<size_t>(
+    "surface_mesh_handle",
+    "Handle to a surface mesh that will represent this object");
+
+  return params;
+}
+
+SurfaceMeshLogicalVolume::SurfaceMeshLogicalVolume(
+  const chi_objects::InputParameters& params)
+  : LogicalVolume(params),
+    surf_mesh(chi::GetStackItemPtrAsType<chi_mesh::SurfaceMesh>(
+      chi::object_stack,
+      params.GetParamValue<size_t>("surface_mesh_handle"),
+      __FUNCTION__)),
+    xbounds_({1.0e6, -1.0e6}),
+    ybounds_({1.0e6, -1.0e6}),
+    zbounds_({1.0e6, -1.0e6})
+{
+  const auto& vertices = surf_mesh->GetVertices();
+  for (auto& vertex : vertices)
+  {
+    const double x = vertex.x;
+    const double y = vertex.y;
+    const double z = vertex.z;
+    if (std::addressof(vertex) == std::addressof(vertices.front()))
+    {
+      xbounds_[0] = x;
+      xbounds_[1] = x;
+      ybounds_[0] = y;
+      ybounds_[1] = y;
+      zbounds_[0] = z;
+      zbounds_[1] = z;
+    }
+    else
+    {
+      xbounds_[0] = std::min(xbounds_[0],x);
+      xbounds_[1] = std::max(xbounds_[1],x);
+      ybounds_[0] = std::min(ybounds_[0],y);
+      ybounds_[1] = std::max(ybounds_[1],y);
+      zbounds_[0] = std::min(zbounds_[0],z);
+      zbounds_[1] = std::max(zbounds_[1],z);
+    }
+  }
+}
```

Comment:
If I am reading this correctly, this is finding the mesh bounds from a collection of vertices. On my previous comment... You are setting xbounds_[0] with a min and xbounds_[1] with a max, whereas in the constructor it was specified as max, then min.